lib/sync-settings.coffee

```diff
@@ -300,6 +301,21 @@ SyncSettings =
         console.debug "config.set #{keyPath[1...]}=#{value}"
         atom.config.set keyPath[1...], value
 
+  uninstallRedundantPackages: (packages, cb) ->
+    available_packages = @getPackages()
+    redundant_packages = []
+    for pkg in available_packages
+      available_package = (p for p in packages when p.name is pkg.name)
+      if available_package.length is 0
+        redundant_packages.push(pkg)
+    if redundant_packages.length is 0
+      atom.notifications.addInfo "Sync-settings: no packages to remove"
+      return cb?()
+    else
+      for pkg in redundant_packages
+        packageManager = new PackageManager()
+        packageManager.uninstall pkg
```

Comment:
This call of `uninstall` should also pass a callback to inform the user about progress / success / failure of the command.